src/rgw/rgw_op.h

```diff
@@ -309,6 +309,8 @@ class RGWOp : public DoutPrefixProvider {
   virtual const char* name() const = 0;
   virtual RGWOpType get_type() { return RGW_OP_UNKNOWN; }
   virtual std::string canonical_name() const { return fmt::format("REST.{}.{}", s->info.method, name()); }
+  // by default we log all bucket operations
+  virtual bool always_log() const { return s->bucket != nullptr; }
```

Comment:
nit: these functions are specific to the 'server access logging' feature so their names should probably reflect that. this could be confused with several other kinds of logging (dout, ops log, usage log, etc)

Reply:
i will rename the new one in this PR.

will change `canonical_name()` to `bucket_logging_name()` in a separate PR after the bulk of bucket logging PRs is merged (to avoid conflicts).
I also want to arrange these functions in one file (as much as possible) instead of different levels in the inheritence, as they are now.